src/Compilers/CSharp/Portable/CodeGen/EmitExpression.cs

```diff
@@ -3466,11 +3466,15 @@ private void EmitCallCleanup(SyntaxNode syntax, UseKind useKind, MethodSymbol me
             }
         }
 
-        private void EmitLoadFunction(BoundFunctionPointerLoad load)
+        private void EmitLoadFunction(BoundFunctionPointerLoad load, bool used)
         {
             Debug.Assert(load.Type is { TypeKind: TypeKind.FunctionPointer });
-            _builder.EmitOpCode(ILOpCode.Ldftn);
-            EmitSymbolToken(load.TargetMethod, load.Syntax, optArgList: null);
+
+            if (used)
+            {
+                _builder.EmitOpCode(ILOpCode.Ldftn);
+                EmitSymbolToken(load.TargetMethod, load.Syntax, optArgList: null);
```

Comment:
>EmitSymbolToken(load.TargetMethod, load.Syntax, optArgList: null); [](start = 16, length = 66)

Can this have a side effect, for example an exception? Is this conditional handling consistent with what we do for delegate creation? I.e. if a delegate instance is not used, are we not loading the address of a target method?

Reply:
If I hadn't fixed the remapped local function bug, it would have an effect, but the only effect from this I could see now is if a tree-trimming algorithm decides the referenced method isn't used (which it isn't) and decides to remove it. However, when I implemented this I checked our handling for delegates:

```cs
static void M()
{
    System.Action a = M;
}
```

produces this IL in release mode:

```
    .method private hidebysig static 
        void M () cil managed 
    {
        // Method begins at RVA 0x2050
        // Code size 1 (0x1)
        .maxstack 8

        IL_0000: ret
    } // end of method C::M
```

---
In reply to: [446421466](https://github.com/dotnet/roslyn/pull/45473#discussion_r446421466) [](ancestors = 446421466)

Reply:
My intuition was that the tree trimming aspect was fine, and likely desired. The fact that it lines up with `delegate` behavior makes me feel good about this outcome.